### ISSUE TITLE
Overloaded methods cannot be called using function call syntax

### ISSUE THREAD

**Issue by jfecher** [noir-lang/noir#3088]
### Aim

Trying to call an overloaded method using function call syntax:
```rs
struct Foo<T> {}

impl Foo<u32> {
    fn foo(self) -> Field { 1 }
}

impl Foo<u64> {
    fn foo(self) -> Field { 2 }
}

fn main() {
    let f: Foo<u32> = Foo{};
    assert(Foo::foo(f) == 1);
}
```

### Expected Behavior

The program to execute without failure

### Bug

```
error: Could not resolve 'foo' in path
   ┌─ /.../short/src/main.nr:15:17
   │
15 │     assert(Foo::foo(f) == 1);
   │                 ---
   │

Aborting due to 1 previous error
exit 1
```

### To Reproduce

1.
2.
3.
4.

### Installation Method

None

### Would you like to submit a PR for this Issue?

No

**Comment by asterite** [comment]
@jfecher I think this was fixed in #3087 but wasn't correctly linked, so we could close this. But let me know if this isn't correct! (the code doesn't seem to trigger the bug anymore)

**Comment by jfecher** [comment]
@asterite this wasn't fixed by #3087 (rather that PR introduced this as a limitation) but it seems this was fixed sometime since then
